sky/backends/cloud_vm_ray_backend.py

```diff
@@ -173,7 +173,10 @@ def add_prologue(self,
             SKY_REMOTE_WORKDIR = {log_lib.SKY_REMOTE_WORKDIR!r}
             job_lib.set_status({job_id!r}, job_lib.JobStatus.PENDING)
 
-            ray.init(address = 'ray://localhost:10001', namespace='__sky__{job_id}__', log_to_driver=True)
+            # Should use 'auto' or 'ray://<internal_head_ip>:10001' rather than 
+            # 'ray://localhost:10001', or 'ray://127.0.0.1:10001'. Otherwise, it will
+            # trigger a bug of ray job failed to get the placement group in ray 2.0.0.
+            ray.init(address='auto', namespace='__sky__{job_id}__', log_to_driver=True)
```

Comment:
> @michaelzhiluo Can you remind me why you change the address from auto to localhost in https://github.com/skypilot-org/skypilot/pull/763 ?

It was because @michaelzhiluo  found that on Ray 1.13, using `auto` broke on-prem, and only ray client works. Discussion: https://github.com/skypilot-org/skypilot/pull/763#discussion_r915967669

I think we can let this PR in first by specializing what to use here based on whether cloud or on-prem.  In a next PR, maybe @michaelzhiluo can investigate whether `auto` works for on-prem on Ray 2.0?